modules/components/analytics/core/generic/org.wso2.carbon.pc.analytics.core.generic/src/main/java/org/wso2/carbon/pc/analytics/core/generic/TaskLevelMonitor.java

```diff
@@ -513,4 +513,67 @@ public String getTaskList() {
 		}
 		return taskIdList;
 	}
+
+	/**
+	 * Get task definition key list for a selected process id
+	 *
+	 * @param filters is used to get process id as a filter
+	 * @return task definition key list as a JSON array string
+     */
+	public String getProcessRelatedTaskList(String filters) {
+		String taskIdList = "";
+		try {
+			if (AnalyticsUtils.isDASAnalyticsActivated()) {
+				JSONObject filterObj = new JSONObject(filters);
+				String processId = filterObj.getString(AnalyticsConstants.PROCESS_ID);
+				AggregateField countField = new AggregateField();
+				countField.setFieldName(AnalyticsConstants.ALL);
+				countField.setAggregate(AnalyticsConstants.COUNT);
+				countField.setAlias(AnalyticsConstants.TASK_INSTANCE_COUNT);
+
+				ArrayList<AggregateField> aggregateFields = new ArrayList<>();
+				aggregateFields.add(countField);
+
+				String queryStr = "processDefinitionId:" + "\"'" + processId+ "'\"";
+				AggregateQuery query = new AggregateQuery();
+				query.setTableName(AnalyticsConstants.USER_INVOLVE_TABLE);
+				query.setGroupByField(AnalyticsConstants.TASK_DEFINITION_KEY);
+				query.setAggregateFields(aggregateFields);
+				query.setQuery(queryStr);
+
+				if (log.isDebugEnabled()) {
+					log.debug("Query to get the Task List Result:" +
+							AnalyticsUtils.getJSONString(query));
+				}
+
+				String result = AnalyticsRestClient
+						.post(AnalyticsUtils.getURL(AnalyticsConstants.ANALYTICS_AGGREGATE),
+								AnalyticsUtils.getJSONString(query));
+
+				JSONArray array = new JSONArray(result);
+				JSONArray resultArray = new JSONArray();
+
+				if (array.length() != 0) {
+					for (int i = 0; i < array.length(); i++) {
+						JSONObject jsonObj = array.getJSONObject(i);
+						JSONObject values = jsonObj.getJSONObject(AnalyticsConstants.VALUES);
+						String taskDefKey =
+								values.getJSONArray(AnalyticsConstants.TASK_DEFINITION_KEY)
+										.getString(0);
+						JSONObject o = new JSONObject();
+						o.put(AnalyticsConstants.TASK_DEFINITION_KEY, taskDefKey);
+						resultArray.put(o);
+					}
+					taskIdList = resultArray.toString();
+				}
+
+				if (log.isDebugEnabled()) {
+					log.debug("Task List Result:" + taskIdList);
+				}
+			}
+		} catch (Exception e) {
```

Comment:
Catching top-level `Exception` is not recommended. You should be more specific. When you catch the exception **you're supposed to handle it properly**.  And you cannot expect to handle all kind of exceptions in your code. read SO answers [here](http://stackoverflow.com/questions/2416316/why-is-the-catchexception-almost-always-a-bad-idea).